lib/ex_unit/lib/ex_unit/capture_log.ex

```diff
@@ -84,24 +84,32 @@ defmodule ExUnit.CaptureLog do
   end
 
   defp add_capture(pid, opts) do
-    GenEvent.add_mon_handler(Logger, {Console, pid}, {pid, opts})
+    parent = self()
+    {:ok, proxy_pid} = Task.start fn ->
+      case :gen_event.add_sup_handler(Logger, {Console, pid}, {pid, opts}) do
+        :ok ->
+          ref = Process.monitor(parent)
+          send parent, {self(), :ok}
+          receive do
+            {:DOWN, ^ref, :process, ^parent, _reason} -> :ok
+            {:gen_event_EXIT, {Console, ^pid}, _reason} -> :ok
+          end
+        other ->
+          send parent, {self(), other}
+      end
+    end
+    receive do
+      {^proxy_pid, result} -> result
```

Comment:
When awaiting a response for a process that is not linked directly or via supervision tree we should monitor the process. This means we know immediately when the failure occurs. I'm not sure a timeout helps here because we want to wait for the `Logger` process to handle the call.